File: backend/src/main/java/com/snack/entity/Notification.java
package com.snack.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("notification")
public class Notification {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    /** EXPIRY_SOON / EXPIRED / LOW_STOCK / STOCK_OUT */
    private String type;

    private String title;

    private String content;

    private Long relatedId;

    private LocalDate notifyDate;

    private Integer isRead;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}

File: frontend/src/types/notification.ts
export interface Notification {
  id: number
  type: string
  title: string
  content: string
  relatedId: number
  snackName: string
  notifyDate: string
  isRead: number
  createTime: string
}

export interface UnreadCount {
  count: number
  expirySoonCount: number
  expiredCount: number
  lowStockCount: number
  stockOutCount: number
}
